lib/silencer/rails/logger.rb

```diff
@@ -45,7 +45,7 @@ def call(env)
       private
 
       def quiet(&block)
-        if ::Rails.logger.respond_to?(:silence)
+        if ::Rails.logger.respond_to?(:silence) && ::Rails.logger.method(:silence).owner != ::Kernel
```

Comment:
In Rails 4 they defined `silence` multiple times, one here: [Kernel#silence](https://github.com/rails/rails/blob/v4.2.7/activesupport/lib/active_support/core_ext/kernel/reporting.rb#L108) (with a mandatory parameter, we don't want to call this), the other one is [LoggerSilence#silence](https://github.com/rails/rails/blob/v4.2.7/activesupport/lib/active_support/logger_silence.rb#L13) (with an optional parameter, we want to call this).

I got the guarding code from this commit: https://github.com/rails/rails/commit/33015969d1e585073709f7803515acf19e3b2d5f